### DISCUSSION TITLE
Is there a way to configure rust-analyzer's "Open Docs" feature?

### DISCUSSION THREAD

**Discussion by andrewbanchich** [joaotavora/eglot#1327] [Q&A] [answered]
https://emacs-lsp.github.io/lsp-mode/page/lsp-rust-analyzer/#open-external-documentation

In `lsp-mode` this opens a web browser with the doc site for the symbol under the cursor.

**Comment by nemethf** [comment, marked as answer]
open-external-documentation uses a non-standard LSP extension, so Eglot does not and most probably will not support it.  But the implementation is straightforward: 

```elisp
(defun eglot-x-open-external-documentation ()
  "Open a URL to the documentation for the symbol under point."
  (interactive)
  (eglot-server-capable-or-lose :experimental :externalDocs)
  (let ((res (jsonrpc-request (eglot--current-server-or-lose)
                              :experimental/externalDocs
                              (eglot--TextDocumentPositionParams))))
    (when res
      (let ((local (plist-get res :local))
            (web (or (plist-get res :web)
                     res)))
        (if (and local (file-exists-p (eglot--uri-to-path local)))
            (browse-url local)
          (browse-url web))))))
```

**Reply by andrewbanchich** [reply]
awesome, thank you so much!

**Reply by joaotavora** [reply]
Awesome.  This could go into the manual's new "Extending Eglot" section, which currently opens with a more complicated example of how to support `clangd`'s `:inactiveRegions` extension.  Opening with this would make a more gentle introduction.  `eglot-uri-to-path` is now public, btw, so you don't need the `--` version there.

**Reply by nemethf** [reply]
I've just copied this from eglot-x.

> `eglot-uri-to-path` is now public, btw, so you don't need the `--` version there.

That's good to know.  Currently, I don't follow the recent developments of Eglot closely.

**Reply by nemethf** [reply]
> > `eglot-uri-to-path` is now public, btw, so you don't need the `--` version there.

It seem that's not yet true:

```elisp
(define-obsolete-function-alias 'eglot--uri-to-path 'eglot-uri-to-path "1.16")
```
But the latest Eglot release is 1.15.
